packages/react-data-grid/src/Row.tsx

```diff
@@ -0,0 +1,151 @@
+
+import rowComparer from './common/utils/RowComparer';
+import Cell from './Cell';
+import { isFrozen } from './ColumnUtils';
+import { Column, RowRenderer, RowRendererProps, CellRenderer, CellRendererProps } from './common/types';
+
+export default class Row extends React.Component<RowRendererProps> implements RowRenderer {
+  static displayName = 'Row';
+
+  static defaultProps = {
+    cellRenderer: Cell,
+    isSelected: false,
+    height: 35
+  };
+
+  private readonly row = React.createRef<HTMLDivElement>();
+  private readonly cells: { [key: string]: CellRenderer | null } = {};
+
+  shouldComponentUpdate(nextProps: RowRendererProps) {
+    return rowComparer(nextProps, this.props);
+  }
+
+  handleDragEnter = (e: React.DragEvent<HTMLDivElement>) => {
+    // Prevent default to allow drop
+    e.preventDefault();
+    const { idx, cellMetaData } = this.props;
+    cellMetaData.onDragEnter(idx);
+  };
+
+  handleDragOver = (e: React.DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    e.dataTransfer.dropEffect = 'copy';
+  };
+
+  handleDrop = (e: React.DragEvent<HTMLDivElement>) => {
+    // The default in Firefox is to treat data in dataTransfer as a URL and perform navigation on it, even if the data type used is 'text'
+    // To bypass this, we need to capture and prevent the drop event.
+    e.preventDefault();
+  };
+
+  getCell(column: Column) {
+    const Renderer = this.props.cellRenderer;
+    const { idx, cellMetaData, isScrolling, row, isSelected, scrollLeft, lastFrozenColumnIndex } = this.props;
+    const { key } = column;
+
+    const cellProps: CellRendererProps & { ref: (cell: CellRenderer | null) => void } = {
+      ref: (cell) => {
+        this.cells[key] = cell;
+      },
+      idx: column.idx!, // TODO: fix idx type
+      rowIdx: idx,
+      height: this.getRowHeight(),
+      column,
+      cellMetaData,
+      value: this.getCellValue(key || String(column.idx)), // TODO: fix idx type
+      rowData: row,
+      isRowSelected: isSelected,
+      expandableOptions: this.getExpandableOptions(key),
+      isScrolling,
+      scrollLeft,
+      lastFrozenColumnIndex
+    };
+
+    return <Renderer key={`${key}-${idx}`} {...cellProps} />;
+  }
+
+  getCells() {
+    const { colOverscanStartIdx, colOverscanEndIdx, columns } = this.props;
+    const frozenColumns = columns.filter((c?: Column) => isFrozen(c!)) as Column[];
+    const nonFrozenColumn = columns.slice(colOverscanStartIdx, colOverscanEndIdx + 1).filter((c?: Column) => !isFrozen(c!)) as Column[];
+    return nonFrozenColumn.concat(frozenColumns)
+      .map((c?: Column) => this.getCell(c!));
+  }
+
+  getRowTop(): number {
+    const { current } = this.row;
+    return current ? current.offsetTop : 0;
+  }
+
+  getRowHeight(): number {
+    return this.props.height;
+  }
+
+  getCellValue(key: string) {
+    let val;
+    if (key === 'select-row') {
+      return this.props.isSelected;
+    }
+    if (typeof this.props.row.get === 'function') {
```

Comment:
I'm guessing `row` can be an immutable map?
```js
if (Immutable.Map.isMap(this.props.row)) {
```

Reply:
I think we can completely get rid of immuatable js and as it is not integrated properly. We are using `rows[column.key]` in a few places. I will revisit this